app/services/shepherd.js

```diff
@@ -0,0 +1,145 @@
+import { schedule, later } from "@ember/runloop";
+import Service, { inject as service } from "@ember/service";
+import { tracked } from "@glimmer/tracking";
+import Tours from "timed/tours";
+
+export default class Shepherd extends Service {
+  @service tour;
+  @service notify;
+  @service media;
+  @service router;
+  @service autostartTour;
+  @tracked model;
+
+  get routeName() {
+    return this.router.get("currentRoute")?.name;
+  }
+  async prepare() {
+    this.addDefaultStepsOptions();
+    await this.startFromBeginning();
+    await this.showToursOfCurrentRoute();
+    this.watchRouteChanges();
+  }
+
+  async showToursOfCurrentRoute() {
+    if (this.isThereStepForThisRoute()) {
+      const steps = this.getRouteTours();
+      await this.addSteps(steps.map(this.toStep));
+    }
+  }
+
+  startFromBeginning() {
+    return new Promise((resolve) => {
+      if (this.routeName !== "index.activities.index") {
+        this.router
+          .transitionTo("index.activities.index")
+          .then(() => resolve());
+        return;
```

Comment:
```suggestion
  async startFromBeginning() {
      if (this.routeName === "index.activities.index") {
          return;
       }
        return await this.router
          .transitionTo("index.activities.index");
```
*This suggestion needs a run through prettier.*